flink-runtime/src/main/java/org/apache/flink/runtime/source/coordinator/SourceCoordinator.java

```diff
@@ -535,22 +547,24 @@ private void runInEventLoop(
 
         context.runInCoordinatorThread(
                 () -> {
-                    try {
-                        action.run();
-                    } catch (Throwable t) {
-                        // if we have a JVM critical error, promote it immediately, there is a good
-                        // chance the
-                        // logging or job failing will not succeed any more
-                        ExceptionUtils.rethrowIfFatalErrorOrOOM(t);
-
-                        final String actionString =
-                                String.format(actionName, actionNameFormatParameters);
-                        LOG.error(
-                                "Uncaught exception in the SplitEnumerator for Source {} while {}. Triggering job failover.",
-                                operatorName,
-                                actionString,
-                                t);
-                        context.failJob(t);
+                    try (MdcUtils.MdcCloseable mdcCloseable =
+                            MdcUtils.withContext(MdcUtils.asContextData(jobID))) {
+                        try {
+                            action.run();
```

Comment:
Same as above, I think it should be possible to wrap the executor inside the context.